lib/queryUtil.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports = {
+    count: function(domain, query, callback) {
+        var domainModel = require('../models/' + domain);
```

Comment:
Instead of doing a `require` here, should probably `require('mongoose')` and use Mongoose to pull in the model via it's schema name, and update the services to use the schema name.. Just make sure that your calling service actually pulls in model with require..
e.g.:
```javascript
var mongoose = require('mongoose');
...

   count: function(name, query, callback) {
       mongoose.model(name).count(query, function(err, count) {
...
```
Same pattern is used in `test/testdata.js`.